### PR TITLE
Product creation using package photo: Media picker library

### PR DESCRIPTION
**Pull request by 0nko**
This PR integrates the Media picker library for selecting an package photo. Supported media sources are: **Local device media**, **device camera** & **WP media library**. The image can be changed, but it is not scanned for keywords (next PR).

[Demo](https://github.com/woocommerce/woocommerce-android/assets/1522856/cf9f813e-58f8-4ad3-b831-a45351178ed9)

**To test:**
1. Go to Products
2. Tap on + FAB
3. Select the AI option
4. Tap on Use package photo button
5. Notice a media picker dialog with 3 options appears
6. Try each one and verify an image is properly displayed in the package photo bottom sheet
7. Once a photo is loaded, tap on the Pencil button above the image
8. Try changing the image and verify it works

*Note:* Please merge #10001 first.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/mediapicker/MediaPickerHelper.kt [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import androidx.fragment.app.Fragment
+import com.woocommerce.android.mediapicker.MediaPickerUtil.processDeviceMediaResult
+import com.woocommerce.android.mediapicker.MediaPickerUtil.processMediaLibraryResult
+import dagger.hilt.android.scopes.FragmentScoped
+import org.wordpress.android.mediapicker.api.MediaPickerSetup
+import org.wordpress.android.mediapicker.api.MediaPickerSetup.DataSource
+import org.wordpress.android.mediapicker.ui.MediaPickerActivity
+import javax.inject.Inject
+
+@FragmentScoped
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/mediapicker/MediaPickerHelper.kt:15]: Unsure this scope might be the reason but this class will be destroyed upon configuration changes such us changing between light/dark mode. 

To reproduce you can open media picker and select "Choose from device". 
Then change app theme to dark or light mode. 
Select a photo
Click on "Use this as media" 

The app will crash: 
```
com.woocommerce.android.dev          E  FATAL EXCEPTION: main
                                                                                                    Process: com.woocommerce.android.dev, PID: 19696
                                                                                                    java.lang.RuntimeException: Unable to resume activity {com.woocommerce.android.dev/com.woocommerce.android.ui.main.MainActivity}: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=1847146516, result=-1, data=Intent { (has extras) }} to activity {com.woocommerce.android.dev/com.woocommerce.android.ui.main.MainActivity}: kotlin.UninitializedPropertyAccessException: lateinit property onMediaSelected has not been initialized
                                                                                                    	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4444)
                                                                                                    	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4476)
                                                                                                    	at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:52)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                                                    	at android.os.Looper.loop(Looper.java:223)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7656)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
                                                                                                    Caused by: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=1847146516, result=-1, data=Intent { (has extras) }} to activity {com.woocommerce.android.dev/com.woocommerce.android.ui.main.MainActivity}: kotlin.UninitializedPropertyAccessException: lateinit property onMediaSelected has not been initialized
                                                                                                    	at android.app.ActivityThread.deliverResults(ActivityThread.java:5015)
                                                                                                    	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4431)
                                                                                                    	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4476) 
                                                                                                    	at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:52) 
                                                                                                    	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176) 
                                                                                                    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97) 
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066) 
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106) 
                                                                                                    	at android.os.Looper.loop(Looper.java:223) 
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7656) 
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method) 
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592) 
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947) 
                                                                                                    Caused by: kotlin.UninitializedPropertyAccessException: lateinit property onMediaSelected has not been initialized
                                                                                                    	at com.woocommerce.android.mediapicker.MediaPickerHelper.handleDeviceMediaResult(MediaPickerHelper.kt:50)
                                                                                                    	at com.woocommerce.android.mediapicker.MediaPickerHelper.deviceLibraryLauncher$lambda$0(MediaPickerHelper.kt:22)
                                                                                                    	at com.woocommerce.android.mediapicker.MediaPickerHelper.$r8$lambda$QD2oHoWoG_d99UlKljMs9cSA6Qg(Unknown Source:0)
                                                                                                    	at com.woocommerce.android.mediapicker.MediaPickerHelper$$ExternalSyntheticLambda0.onActivityResult(Unknown Source:4)
                                                                                                    	at androidx.activity.result.ActivityResultRegistry.doDispatch(ActivityResultRegistry.java:418)

```


https://github.com/woocommerce/woocommerce-android/assets/2663464/0c07c255-af2d-4473-8c8b-46a9cdf2380a


Btw, unrelated to this PR but looks like we have a bug in the `Play`button of the media picker in Light mode. I'll create an issue for it.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/mediapicker/MediaPickerHelper.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    private val mediaLibraryLauncher = fragment.registerForActivityResult(StartActivityForResult()) {
+        handleMediaLibraryPickerResult(it)
+    }
+
+    private fun showMediaPicker(source: DataSource) {
+        val mediaPickerIntent = MediaPickerActivity.buildIntent(
+            context = fragment.requireContext(),
+            mediaPickerSetupFactory.build(source)
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/mediapicker/MediaPickerHelper.kt]: I'd suggest setting `MediaTypes` here to `IMAGES` only. Enabling everything will enable users to select a video media file, which won't work: 

https://github.com/woocommerce/woocommerce-android/assets/2663464/9e162756-ecaa-4300-a6f1-0fe5fbef91fc

**Comment by JorgeMucientes** [comment]
Excellent job @0nko! Works well for the most part but there are a couple issues I believe we should address before merging. Please let me know what you think 🙂.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/mediapicker/MediaPickerHelper.kt (hunk 1 above)
Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/mediapicker/MediaPickerHelper.kt:15]: Good catch!

File: WooCommerce/src/main/kotlin/com/woocommerce/android/mediapicker/MediaPickerHelper.kt (hunk 2 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/mediapicker/MediaPickerHelper.kt]: @0nko just for more context on why this is needed now, as part of a Hack Week and in order to support different file types for the product downloadable files, I made some changes to the MediaPicker library, and this updates the `build` function of the factory with an argument to specify the file types, by default it allows all file types, the PR where the changes were made is here: https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/pull/71

Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/mediapicker/MediaPickerHelper.kt]: 👍

**Comment by 0nko** [comment]
@JorgeMucientes Thanks for the review! I've addressed the issues, ready for another round.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/mediapicker/MediaPickerHelper.kt (hunk 2 above)
Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/mediapicker/MediaPickerHelper.kt]: @hichamboushaba Thanks for the context!

**Review by JorgeMucientes — APPROVED** [review]

**Comment by JorgeMucientes** [comment]
Thanks for tackling the changes @0nko Works like a charm 👍🏼 !

**Pull request merged**